api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1701,12 +1701,13 @@ type AWSServiceEndpoint struct {
 // would be pre-created and then their names would be used respectively in the ResourceGroupName, SubnetName, VnetName
 // fields of the Hosted Cluster CR. An existing cloud resource is expected to exist under the same SubscriptionID.
 type AzurePlatformSpec struct {
-	// Credentials is the object containing existing Azure credentials needed for creating and managing cloud
-	// infrastructure resources.
+	// TenantID is the id for the Azure tenant in which this Hosted Cluster will be created.
 	//
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="TenantID is immutable"
+	// +immutable
 	// +required
-	Credentials corev1.LocalObjectReference `json:"credentials"`
+	TenantID string `json:"tenantID,omitempty"`
```

Comment:
FWIW a quick review of the forums, etc, shows that this (and the subscription ID) are not secret values.